### PR TITLE
v0.5.0 release candidate branch

### PR DESCRIPTION
**Pull request by rhornung67**
# Summary

- This PR is for the v0.5.0 release candidate.
- Please check your committed PRs against the draft release notes and update the release notes to capture all notable changes that users should be aware of. Check the box next to your name below when you have verified that the release notes properly account for your contributions:

- [x] @white238 
- [ ] @cyrush 
- [x] @kennyweiss 
- [x] @nselliott 
- [ ] @ltaylor16 
- [x] @agcapps 
- [x] @bmhan12 
- [x] @joshessman-llnl 
- [x] @rhornung67

### CODE REVIEW

**Comment by agcapps** [comment]
In RELEASE-NOTES.md, there are lists of "Added," "Fixed," and "Changed" items.  Do newest items go on the top of these lists, or the bottom?  I have one item to add, and I'd like to put it (close to) the right place in the list.

**Review by agcapps — APPROVED** [review]
Noted the one change I made

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -497,16 +500,19 @@ The Axom project release numbers follow [Semantic Versioning](http://semver.org/
 ### Known Bugs
 -
 
-[Unreleased]:    https://github.com/LLNL/axom/compare/v0.4.0...develop
+[Unreleased]:    https://github.com/LLNL/axom/compare/v0.5.0...develop
+[Version 0.5.0]: https://github.com/LLNL/axom/compare/v0.4.0...v0.5.0
```
Comment (kennyweiss) [inline RELEASE-NOTES.md:504]: Note: This link will only work after we create the `v0.5.0` tag

**Review by kennyweiss — APPROVED** [review]

**Review by joshessman-llnl — APPROVED** [review]

**Comment by ltaylor16** [comment]
I was going to merge develop into this branch then start the release process.
Any objections?

**Comment by rhornung67** [comment]
@ltaylor16 It looks like all the PRs we wanted to have in have been merged. So, unless someone objects, I think you can merge develop into this branch. Before starting the release @nselliott can you double-check the release notes to make sure they include the SCR updates. Also, @white238 @kennyweiss if there are any build system changes that users should know about based on a newer version of BLT, please make sure that they are captured in the release notes.

**Review by nselliott — APPROVED** [review]

**Comment by rhornung67** [comment]
Thanks @nselliott !

**Comment by rhornung67** [comment]
@ltaylor16 when you change this PR to be merged into main for the release (next step), there will be a huge number of modifications since main hasn't changed since our last release. To make it easier to review a release candidate PR like that, I usually make a corresponding PR to merge the RC branch into develop and mention it in the PR to merge the RC branch into main. It will only contain the changes related to the release prep. Tell folks to review the changes in the PR to merge into develop, but approve the PR to merge the RC branch into main. There are other ways to do this as well, but this one seems the easiest to me.

**Pull request merged**